xdsl/passes.py

```diff
@@ -101,9 +101,13 @@ def from_pass_spec(cls, spec: PipelinePassSpec) -> Self:
                 raise ValueError(f'Pass {cls.name} requires argument "{op_field.name}"')
 
             # convert pass arg to the correct type:
+            field_type = op_field.type
+            if isinstance(field_type, str):
+                # If the type was provided as a string, eval the string to get the type
+                field_type = eval(field_type, None)
```

Comment:
Instead of an `eval` here, can we use `field_types = typing.get_type_hints(cls)` before the loop, and then use `field_types[op_field.name]` to get the actual type?
That should be nicer than calling `eval`? I'm not sure there would be too much of a runtime cost from that?

Reply:
Sure I'll refactor it. It does just call eval for you but I agree that it's nicer